webapp/src/components/dataset_input/FileSelection.vue

```diff
@@ -0,0 +1,171 @@
+          >
+            select {{ fileType }}
+          </span>
+          <input
+            ref="inputFileElement"
+            type="file"
+            :multiple="multiple"
+            :accept="acceptFilter"
+            class="hidden"
+            @change="async () => await submitFiles()"
+          />
+        </label>
+      </div>
+      <!-- Display what has been connected -->
+      <div
+        v-if="model === undefined"
+        class="flex justify-end items-center mt-1"
+      >
+        <span
+          class="hover:cursor-pointer"
+          v-tippy="{
+            content: 'Data always stays on your device and is never shared.',
+          }"
+        >
+          <i class="fa fa-lock mr-1" />
+        </span>
+      </div>
+
+      <!-- Display some text if specified -->
+      <div
+        v-if="$slots.default && model === undefined"
+        class="flex justify-center mt-5"
+      >
+        <p class="text-slate-500 text-sm">
+          <span><slot /></span>
+        </p>
+      </div>
+
+      <!-- If only one file is connected, display its name, if multiple display the number of files -->
+      <div
+        v-if="model !== undefined"
+        class="pt-4 flex flex-col items-center pb-5"
+      >
+        <div
+          class="mb-4 flex justify-center items-center text-center md:text-left sm:text-lg text-disco-blue"
+        >
+          <span v-if="multiple"
+            >Number of selected files:
+            <span class="pl-1 text-xl">{{ model.size }}</span></span
+          >
+          <span v-else class="pl-1">{{ model.first()?.name ?? "none" }}</span>
+        </div>
+        <div>
+          <CustomButton @click="clearFiles">
+            clear {{ fileType }}
+          </CustomButton>
+        </div>
+      </div>
+    </section>
+  </article>
+</template>
+
+<script lang="ts" setup>
+import { Set } from "immutable";
+import { ref, computed } from "vue";
+
+import CustomButton from "@/components/simple/CustomButton.vue";
+
+const props = withDefaults(
+  defineProps<{
+    type: "tabular" | "image" | "text";
+    multiple?: boolean; // accept one or multiple files
+  }>(),
+  {
+    multiple: false,
+  },
+);
+
+const model = defineModel<Set<File>>();
+
+const inputFileElement = ref<HTMLInputElement | null>(null);
+
+const hideConnectField = computed(() => model.value !== undefined);
+
+const fileType = computed(() => {
+  const name = (() => {
+    // need function wrap vuejs/eslint-plugin-vue#2142
+    switch (props.type) {
+      case "tabular":
+        return "CSV";
+      case "image":
+        return "image";
+      case "text":
+        return "text";
+    }
+  })();
+
+  return `${name}${props.multiple ? "s" : ""}`;
+});
+const acceptFilter = computed(() => {
+  switch (props.type) {
+    case "tabular":
+      return ".csv";
+    case "image":
+      return "image/*";
+    case "text":
+      return "text/plain";
```

Comment:
It seems I can upload an image or a csv in a text field without any error (and vice-versa), is it expected?

Reply:
hum, on firefox ESR, when it opens the file upload modal, it filters the shown dir with "Text File" (or "Image Files" , …). one can choose to "show all files" and in that case, select anything.
so yeah, you can upload anything but you'll have to actively do it wrong with my setup. is it not the case in your browser?

if we want to check that the files are indeed valid, we need to read the whole dataset to know that (which is not currently done with `datasetBuilder`, only partially when building it). I can add such check in #682 but yeah, garbage-in garbarge-out.

Reply:
Oh I was drag and dropping files so there was no modal to prevent me from connecting wrong things